### PR TITLE
PushCommand-Fixes: Added a lot of tests to cover the following scenarios

### PR DESCRIPTION
**Pull request by deepakaravindr**
1. API v3 index without push endpoint
2. API v3 index not available
3. APIKey for push is provided as a switch
4. APIKey for push is provided via config
5. DefaultPushSource is specified via config

Scenario 1 and 4 failed. Fixed them

@yishaigalatzer @emgarten @MeniZalzman @feiling

### CODE REVIEW

File: src/NuGet.CommandLine/Commands/PushCommand.cs [hunk 1]
```diff
@@ -35,11 +35,24 @@ public class PushCommand : Command
...
             // Don't push symbols by default
             string source = ResolveSource(packagePath, ConfigurationDefaults.Instance.DefaultPushSource);
+            var pushEndpoint = await GetPushEndpointAsync(source);
 
-            var apiKey = GetApiKey(source);
-            if (String.IsNullOrEmpty(apiKey) && !IsFileSource(source))
+            if (string.IsNullOrEmpty(pushEndpoint))
             {
-                Console.WriteWarning(LocalizedResourceManager.GetString("NoApiKeyFound"), CommandLineUtility.GetSourceDisplayName(source));
+                var message = string.Format(
```
Comment (emgarten) [inline src/NuGet.CommandLine/Commands/PushCommand.cs:42]: Not really needed for a single arg, but add CultureInfo here to avoid build warnings.

**Comment by emgarten** [comment]
Great tests.
:shipit:

File: src/NuGet.CommandLine/Commands/PushCommand.cs [hunk 2]
```diff
@@ -48,10 +61,9 @@ public class PushCommand : Command
                 timeout = TimeSpan.FromMinutes(5); // Default to 5 minutes
             }
 
-            source = await GetPushEndpointAsync(source);
-            PushPackage(packagePath, source, apiKey, timeout);
+            PushPackage(packagePath, pushEndpoint, apiKey, timeout);
 
-            if (source.Equals(NuGetConstants.DefaultGalleryServerUrl, StringComparison.OrdinalIgnoreCase))
+            if (pushEndpoint.Equals(NuGetConstants.DefaultGalleryServerUrl, StringComparison.OrdinalIgnoreCase))
```
Comment (MeniZalzman) [inline src/NuGet.CommandLine/Commands/PushCommand.cs:66]: It seems that we will always push the symbols if the user is using the nuget.org source, it feels to as a bug that we are not respecting the arguments

**Comment by MeniZalzman** [comment]
:shipit:

File: src/NuGet.CommandLine/Commands/PushCommand.cs (hunk 2 above)
Comment (yishaigalatzer) [inline src/NuGet.CommandLine/Commands/PushCommand.cs:66]: Is this a bug or not? Not sure why there is a signoff here?

Comment (deepakaravindr) [inline src/NuGet.CommandLine/Commands/PushCommand.cs:66]: Apparently, this is just a legacy behavior, we never broke. I can file a bug if we want. IMO, we should not fix it for this release

**Comment by deepakaravindr** [comment]
Merged with https://github.com/NuGet/NuGet.PackageManagement/commit/9920cbac5e9cf31dce5869d82a6467f00737efe3. Will close once all the open comments are answered

File: src/NuGet.CommandLine/Commands/PushCommand.cs (hunk 1 above)
Comment (deepakaravindr) [inline src/NuGet.CommandLine/Commands/PushCommand.cs:42]: It is not a single arg and it was a long line. So, I have broken it.
LocalizedResourceManager.GetString adds the CultureInfo. So, it should work

**Pull request closed without merging**